account_manager/src/validator/import.rs

```diff
@@ -0,0 +1,231 @@
+            CONFIG_FILENAME
+        );
+
+        let password_opt = loop {
+            eprintln!("");
+            eprintln!("{}", PASSWORD_PROMPT);
+
+            let password = read_password_from_user(stdin_password)?;
+
+            if password.as_ref().is_empty() {
+                eprintln!("Continuing without password.");
+                sleep(Duration::from_secs(1)); // Provides nicer UX.
+                break None;
+            }
+
+            match keystore.decrypt_keypair(password.as_ref()) {
+                Ok(_) => {
+                    eprintln!("Password is correct.");
+                    eprintln!("");
+                    sleep(Duration::from_secs(1)); // Provides nicer UX.
+                    break Some(password);
+                }
+                Err(eth2_keystore::Error::InvalidPassword) => {
+                    eprintln!("Invalid password");
+                }
+                Err(e) => return Err(format!("Error whilst decrypting keypair: {:?}", e)),
+            }
+        };
+
+        // The keystore is placed in a directory that matches the name of the public key. This
+        // provides some loose protection against adding the same keystore twice.
+        let dest_dir = validator_dir.join(format!("0x{}", keystore.pubkey()));
+        if dest_dir.exists() {
+            return Err(format!(
+                "Refusing to re-import an existing public key: {:?}",
+                src_keystore
+            ));
+        }
+
+        fs::create_dir_all(&dest_dir)
+            .map_err(|e| format!("Unable to create import directory: {:?}", e))?;
+
+        // Retain the keystore file name, but place it in the new directory.
+        let dest_keystore = src_keystore
+            .file_name()
+            .and_then(|file_name| file_name.to_str())
+            .map(|file_name_str| dest_dir.join(file_name_str))
+            .ok_or_else(|| format!("Badly formatted file name: {:?}", src_keystore))?;
+
+        // Copy the keystore to the new location.
+        fs::copy(&src_keystore, &dest_keystore)
+            .map_err(|e| format!("Unable to copy keystore: {:?}", e))?;
+
+        // Attempt to make the move atomic in the case where the copy succeeds but the remove
+        // fails.
+        if let Err(e) = fs::remove_file(&src_keystore) {
+            if src_keystore.exists() {
+                // If the original keystore path still exists we can delete the copied one.
+                //
+                // It is desirable to avoid duplicate keystores since this is how slashing
+                // conditions can happen.
+                fs::remove_file(dest_keystore)
+                    .map_err(|e| format!("Unable to remove copied keystore: {:?}", e))?;
+                return Err(format!("Unable to delete {:?}: {:?}", src_keystore, e));
+            } else {
+                return Err(format!("An error occurred whilst moving files: {:?}", e));
+            }
+        }
```

Comment:
I found the fact that the importer removes the original keystore quite counter-intuitive, would be interested to hear thoughts on this @AgeManning, @kirk-baird

Reply:
So here we have generated a bunch of keystores and put them in `keystore_dir`. We then wish to move them to our `validator_dir` such that they'll do validating. Then no longer need the original `keystore_dir`?

If I am reading this correctly the name of the destination directory is the public key. However, it's set to error if that director already exists.
If we don't delete the `src_keystore` and run the command again we will error and exit. 
So the way it's designed it must act as an atomic move rather than copy else re-running the command will result in error.

This does make me a little nervous deleting the destination in the case of an error (when trying to delete the source). Cause if we error but delete half the file and have `src_keystore.exists()` we kill both copies.

I would lean more towards doing a skip if the destination already exists rather than erroring and exiting. Then we would not delete the source. And since the destination is the same name as the public key we should not have duplicates.

Reply:
Yeah it is kind of counter-intuitive deleting any keystore. I see the reasoning. 

When I import a keystore from open-ethereum into geth or vica versa, the original keystore is never modified. 
I think we should not touch the original keystore from an `--import` command. So I'd opt to leave it as is (even if it can potentially cause a slashing). 

In either case we should report to the user. "DO NOT USE THE ORIGINAL KEYSTORE TO AVOID SLASHING"
OR
"THIS WILL MOVE THE KEYSTORE TO THE VALIDATOR DIRECTORY"

I favour the prior

Reply:
Yep, sounds good, changing it now.

Reply:
Updated, it now produces output like the following:

```
WARNING: DO NOT USE THE ORIGINAL KEYSTORES TO VALIDATE WITH ANOTHER CLIENT, OR YOU WILL GET SLASHED.

Keystore found at "/home/michael/Desktop/import/keystore-m_12381_3600_0_0_0-1595827134.json":

 - Public key: 0x992517c69e47fae155359ad0e44e4c269bdd471ba5d7f45426545f6bfe5b1ffc6fb048ef2b088246191836e9b8370ebc
 - UUID: 54a4e865-9c97-4b2f-bab6-a86c2ef60602

If you enter the password it will be stored in validator_definitions.yml so that it is not required each time the validator client starts.

Enter the keystore password, or press enter to omit it:

Password is correct.

Skipping import of keystore for existing public key: "/home/michael/Desktop/import/keystore-m_12381_3600_0_0_0-1595827134.json"

Successfully imported 0 validators (1 skipped).

WARNING: DO NOT USE THE ORIGINAL KEYSTORES TO VALIDATE WITH ANOTHER CLIENT, OR YOU WILL GET SLASHED.
```